C7GameData/RulesFormat.cs

```diff
@@ -5,11 +5,32 @@ namespace C7GameData
     (e.g. in a mod) or as part of the save file. At least for now.
 */
 {
-    using System;
-    public class C7RulesFormat {
-        public string Version { get; set; }
-        public C7RulesFormat() {
-            Version = "v0.0early-prototype";
-        }
-    }
+	using System.Collections.Generic;
+	using System.Text.Json.Serialization;
+
+	public class C7RulesFormat {
+		public string Version { get; set; }
+
+		public List<ExperienceLevel> experienceLevels = new List<ExperienceLevel>();
```

Comment:
Calling out @JimOfLeisure 's opinion on this, as he added this file back in November, and would know more about its intended use than I do.  The commit comment doesn't clarify it enough.

My first thought is that this variable should be in GameData.cs, like all of our other game data.

Of course that leaves the question of, "what is C7RulesFormat"?  Previously, it had just been a placeholder of a version string.  Maybe that's intentional?  Maybe it is meant to correspond to the RULE tab from Civ3 (which we haven't really added yet).

Overall though, I don't like that experienceLevels is over in left field here, away from everything else in GameData.  It's not obvious and there's not a clear reason for it.

I see the comment about comments indicating that's what this is here for; are there comments for that beyond the one at the start of this file?  Unless I'm missing something, I'd rather follow precedent than one comment.

Reply:
Perhaps also relevant, IMO there shouldn't be anything wrong with adding "new" items to our C7 data classes, including those linked from GameData, where appropriate.  promotionChanceAfterDefending/promotionChanceAfterAttacking being good examples of items that aren't moddable in Civ but should be in C7, and should be written out to the JSON save file.

They probably should be in some sort of "Rules" file, but I'd rather have the rules be part of the game data than separte from it.

Reply:
I'll admit I haven't been paying much attention to the work you guys have been doing on saving to JSON. When I saw those comments on C7RulesFormat I just assumed they were recent and relevant. I figured the point of separating out the rules was to allow C7 save files to omit them and instead refer to a default rules file, like in the original game. If the plan is that all saves will have their rules embedded then I don't see any reason to keep the rules separate from the rest of the game data, and agree they should be merged.

Another reason to combine them is that the distinction between game data and rules is unclear in general and we might draw the line in the wrong place. In particular I'm thinking of space 4X games where it's common to feature a ship designer, meaning unit types would need to be part of the game data instead of the immutable rules.

Reply:
I agree that there is a not a clear-cut distinction between game data and rules.

IMO (and @WildWeazel can correct me if his understanding is different), it would be preferable to embed all the rules within the JSON, at least as much as feasible.  "As feasible" meaning that there will likely still have to be call-outs to mod files - custom graphics, AI routines, etc.  But by embedding as much of the rules as we can, we maximize the self-contained aspect of the save file.

Reply:
I created an issue for C7RulesFormat here: #247. I'm going to merge it into C7GameData unless someone swoops in with a good reason to keep it around. But I'd like to first merge this and my other PRs for Combat Part II, then worry about C7RulesFormat.